CMakeLists.txt

```
@@ -8,6 +8,13 @@ set (alldirs
   GEOSwgcm_GridComp
   )
 
+option(BUILD_WITH_GIGATRAJ "Build GEOSgcm with Gigatraj" ON)
+
+if (BUILD_WITH_GIGATRAJ)
+  add_definitions (-DHAS_GIGATRAJ)
```

This is a bit of an anti pattern for CMake.  It should get attached to the target that needs the definition.   Possibly `${this}`

---

@mathomp4 Do you know how to address this comment?

---

Yes, you'd probably want to do:
```suggestion
  target_compile_definitions (${this} PRIVATE HAS_GIGATRAJ)
```

---

Note @weiyuan-jiang You might also need to do this in AGCM and SuperDyn GCs as well. `add_definitions` is pretty powerful as it propagates *everywhere*. The `target_` version is more targeted.